File: Drivo/src/entities/LecturerEntity.ts
import { LectureEntity } from "./LectureEntity"
import { StudentsGroupEntity } from "./StudentsGroupEntity"
import { UserEntity } from "./user.entity"

export class LecturerEntity extends UserEntity {
    public studentsGroups: Array<StudentsGroupEntity> | null;
    public lectures: Array<LectureEntity> | null;
}

File: Drivo.Entities/LecturerEntity.cs
namespace Drivo.Entities
{
    public class LecturerEntity : UserEntity
    {
        public LecturerEntity() : base()
        {

        }

        public LecturerEntity(string userName, string email, string firstName, string lastName, DateTime birthDate, string phoneNumber) : base(userName, email, firstName, lastName, birthDate, phoneNumber)
        {

        }

        public virtual List<StudentsGroupEntity>? StudentsGroups { get; set; }
        public virtual List<LectureEntity>? Lectures { get; set; }
    }
}
